bin/RegexForTermsFrequency.py

```diff
@@ -89,11 +99,21 @@ def refresh_dicos():
             curr_set = top_termFreq_setName_day[0] + str(timestamp)
             content = Paste.Paste(filename).get_p_content()
 
-            #iterate the word with the regex
+            # iterate the word with the regex
             for regex_str, compiled_regex in dico_regex.items():
-                matched = compiled_regex.search(content)
 
-                if matched is not None: #there is a match
+                signal.alarm(PROCESS_TIMEOUT)
+                try:
+                    matched = compiled_regex.search(content)
+                except TimeoutException:
+                    log_msg = "{0} processing timeout".format(filename)
+                    print (log_msg)
+                    publisher.critical(log_msg)
```

Comment:
None of the module are using the log level ``critical`` as it will be common to have timeout on regex matching.
I suggest to replace it by ``info`` or ``debug``.